src/test/script_standard_tests.cpp

```diff
@@ -268,12 +276,32 @@ BOOST_AUTO_TEST_CASE(script_standard_ExtractDestination)
     BOOST_CHECK(ExtractDestination(s, address));
     BOOST_CHECK(std::get<WitnessV0ScriptHash>(address) == scripthash);
 
+    // TxoutType::WITNESS_V1_TAPROOT
+    s.clear();
+    auto xpk = XOnlyPubKey(pubkey);
+    s << OP_1 << ToByteVector(xpk);
+    BOOST_CHECK(ExtractDestination(s, address));
+    BOOST_CHECK(std::get<WitnessV1Taproot>(address) == WitnessV1Taproot(xpk));
+
+    // TxoutType::ANCHOR
+    s.clear();
+    s << OP_1 << ANCHOR_BYTES;
+    BOOST_CHECK(ExtractDestination(s, address));
+    BOOST_CHECK(std::get<PayToAnchor>(address) == PayToAnchor());
+
     // TxoutType::WITNESS_UNKNOWN with unknown version
+    // -> segwit version 1 with an undefined program size (33 bytes in this test case)
```

Comment:
nit: This might be more precise & helpful?
It took me a few minutes to re-grok the difference between the bytes derived from `XOnlyPubKey` vs compressed `CPubKey`.
```suggestion
    // -> segwit version 1 with an non-standard program size
    // (CPubKey::COMPRESSED_SIZE = 33 bytes in this test case)
```